lib/dl_api_lib/dl_api_lib/pivot/native/sorter.py

```diff
@@ -94,7 +107,62 @@ def _single_axis_sort(self, axis: SortAxis, directions: Sequence[OrderDirection]
 
     def _sort_by_measure(self, axis: SortAxis, sorting_piid: int, settings: PivotMeasureSortingSettings) -> None:
-        pass  # TODO: Not implemented yet. Feeling bored? Go right ahead and contribute!
+        sorting_dim_values: Optional[tuple[DataCellVector, ...]] = None
+        other_axis = {SortAxis.columns: SortAxis.rows, SortAxis.rows: SortAxis.columns}[axis]
+
+        for idx, header in enumerate(self._pivot_dframe.iter_axis_headers(axis)):
+            if header.compare_sorting_settings(settings):
+                if sorting_dim_values is None:
+                    sorting_dim_values = header.values
+                    header.info.sorting_direction = settings.direction
+                else:  # should never actually occur, as header_values + role_spec uniquely identify sorting_idx
+                    raise exc.PivotSortingRowOrColumnIsAmbiguous()
+        if sorting_dim_values is None:
+            raise exc.PivotSortingRowOrColumnNotFound()
+
+        # First order the measure values in the selected row/column
+        # (together with their original position indices)
+        indexed_values = [
+            (old_pos_idx, value)
+            for old_pos_idx, value in enumerate(
+                self._native_dframe.iter_values_along_axis(
+                    axis=other_axis,
+                    dim_values=sorting_dim_values,
+                )
+            )
+        ]
+
+        def normalize_measure_value(value: Optional[DataCellVector]) -> Any:
+            """Convert key to a value that can be sorted natively"""
+            normalizer = self._measure_sort_strategy.get_normalizer(
+                pivot_item_id=sorting_piid, direction=settings.direction
+            )
+            return normalizer.normalize_vector_value(value)
+
+        indexed_values.sort(key=lambda pair: normalize_measure_value(pair[1]))  # pair = (old_pos_idx, value)
+
+        reverse = settings.direction == OrderDirection.desc
+
+        # Now reorder the keys in accordance with the sorted measures
+        key_list = self._get_key_list_for_axis(other_axis)
+        key_to_new_idx = {
+            key_list[old_pos_idx]: new_pos_idx for new_pos_idx, (old_pos_idx, _) in enumerate(indexed_values)
+        }
+        key_list.sort(key=lambda key_value: key_to_new_idx[key_value], reverse=reverse)
+
+        if self._has_total(other_axis):  # hack for totals
+            # Find the key
+            total_key: Optional[FlatPivotDataKey] = None
+            for dim_header in self._pivot_dframe.iter_axis_headers(other_axis):
+                if dim_header.info.role_spec.role == PivotHeaderRole.total:
+                    total_key = FlatPivotDataKey(values=dim_header.values)
+
+            assert total_key is not None
+
+            # Manually put the total at the end
+            total_pos = key_list.index(total_key)
+            del key_list[total_pos]
+            key_list.append(total_key)
```

Comment:
you sure it's `total_key`, not `key_list[total_pos]`?

Reply:
Well, the very point of this is to remove the total from its current position and put it in the end.

Reply:
yep, my bad